packages/opencrvs/test/Adaptor.test.js

```diff
@@ -0,0 +1,196 @@
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      request('POST', '/notification', {
+        resourceType: 'Bundle',
+        type: 'document',
+        meta: {
+          lastUpdated: new Date().toISOString(),
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('post', () => {
+  it('makes a post request to create a birth notification', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      post('/notification', {
+        resourceType: 'Bundle',
+        type: 'document',
+        meta: {
+          lastUpdated: new Date().toISOString(),
+        },
+        entry: birthRecordData,
+      })
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('createBirthRecord', () => {
+  it('successfully creates a birth record', async () => {
+    testServer
+      .intercept({
+        path: '/notification',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        resourceType: 'Bundle',
+        type: 'document',
+        entry: birthRecordData,
+      });
+    state.data = birthRecordData;
+
+    const finalState = await execute(
+      createBirthNotification(state => state.data)
+    )(state);
+
+    expect(finalState.data).to.eql({
+      resourceType: 'Bundle',
+      type: 'document',
+      entry: birthRecordData,
+    });
+  });
+});
+
+describe('queryEvents', () => {
+  it('successfully searches for an event', async () => {
+    testServer
+      .intercept({
+        path: '/graphql',
+        method: 'POST',
+        headers: {
+          Authorization: 'Bearer fake-token',
+          'Content-type': 'application/json',
+        },
+      })
+
+      .reply(200, {
+        data: {
+          searchEvents: {
+            totalItems: 0,
+            results: [],
+            __typename: 'EventSearchResultSet',
+          },
+        },
+      });
+
+    const finalState = await execute(
+      queryEvents({
```

Comment:
There is no assertion here that these query parameters are actually used. 

The test isn't strict enough. It's not really testing very much. It doesn't test for example that the `eventLocationId` parameter is actually mapped in the query (you could delete the query and the test would still pass)

Is there any way you can test what is sent to the server? Or how the query is assembled?